pkg/iter/iterator.go

```diff
@@ -166,7 +166,9 @@ func (i *heapIterator) Next() bool {
 
 	// Find in entry which occurs most often which, due to quorum based
 	// replication, is guaranteed to be the correct next entry.
-	i.currEntry = mostCommon(tuples).Entry
+	curr := mostCommon(tuples)
+	i.currEntry = curr.Entry
+	i.currLabels = curr.EntryIterator.Labels()
```

Comment:
This can be done in `NewHeapIterator`, as there is one heap iterator per stream and labels are constant in a stream.